Tools/autotest/arducopter.py

```diff
@@ -6679,77 +6715,89 @@ def get_average_esc_frequency(self):
     def DynamicNotches(self):
         """Use dynamic harmonic notch to control motor noise."""
         self.progress("Flying with dynamic notches")
+        self.context_push()
 
-        self.set_parameters({
-            "AHRS_EKF_TYPE": 10,
-            "INS_LOG_BAT_MASK": 3,
-            "INS_LOG_BAT_OPT": 0,
-            "INS_GYRO_FILTER": 100, # set the gyro filter high so we can observe behaviour
-            "LOG_BITMASK": 958,
-            "LOG_DISARMED": 0,
-            "SIM_VIB_MOT_MAX": 350,
-            "SIM_GYR1_RND": 20,
-        })
-        self.reboot_sitl()
+        ex = None
+        try:
+            self.set_parameters({
+                "AHRS_EKF_TYPE": 10,
+                "INS_LOG_BAT_MASK": 3,
+                "INS_LOG_BAT_OPT": 0,
+                "INS_GYRO_FILTER": 100, # set the gyro filter high so we can observe behaviour
+                "LOG_BITMASK": 958,
+                "LOG_DISARMED": 0,
+                "SIM_VIB_MOT_MAX": 350,
+                "SIM_GYR1_RND": 20,
+            })
+            self.reboot_sitl()
 
-        self.takeoff(10, mode="ALT_HOLD")
+            self.takeoff(10, mode="ALT_HOLD")
 
-        # find a motor peak
-        freq, hover_throttle, peakdb = self.hover_and_check_matched_frequency_with_fft(-15, 200, 300)
+            # find a motor peak
+            freq, hover_throttle, peakdb = self.hover_and_check_matched_frequency_with_fft(-15, 200, 300)
 
-        # now add a dynamic notch and check that the peak is squashed
-        self.set_parameters({
-            "INS_LOG_BAT_OPT": 2,
-            "INS_HNTCH_ENABLE": 1,
-            "INS_HNTCH_FREQ": freq,
-            "INS_HNTCH_REF": hover_throttle/100.,
-            "INS_HNTCH_HMNCS": 5, # first and third harmonic
-            "INS_HNTCH_ATT": 50,
-            "INS_HNTCH_BW": freq/2,
-        })
-        self.reboot_sitl()
+            # now add a dynamic notch and check that the peak is squashed
+            self.set_parameters({
+                "INS_LOG_BAT_OPT": 2,
+                "INS_HNTCH_ENABLE": 1,
+                "INS_HNTCH_FREQ": freq,
+                "INS_HNTCH_REF": hover_throttle/100.,
+                "INS_HNTCH_HMNCS": 5, # first and third harmonic
+                "INS_HNTCH_ATT": 50,
+                "INS_HNTCH_BW": freq/2,
+            })
+            self.reboot_sitl()
 
-        freq, hover_throttle, peakdb1 = \
-            self.hover_and_check_matched_frequency_with_fft(-10, 20, 350, reverse=True)
+            freq, hover_throttle, peakdb1 = \
+                self.hover_and_check_matched_frequency_with_fft(-10, 20, 350, reverse=True)
 
-        # now add double dynamic notches and check that the peak is squashed
-        self.set_parameter("INS_HNTCH_OPTS", 1)
-        self.reboot_sitl()
+            # now add double dynamic notches and check that the peak is squashed
```

Comment:
Something has gone very wrong in here - I think you might have had a conflict and inadvertently reverted one of my recent patches.

You should be able to find your patch easily enough - it will just be adding a single line into a list of i2c rangefinders to test.